# **Data Science Project 2 : Detection of Parkinson's Disease**

## **Topic : Supervised Machine Learning : Classification**

## **Algorithm used : XGBoost Classification Algorithm**

# **--------------------------------------------------------------------------------**

### **Library imports**

In [50]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

### **Reading the csv file and creating a dataframe**

In [51]:
df=pd.read_csv("https://raw.githubusercontent.com/chaitanyabaranwal/ParkinsonAnalysis/master/parkinsons.csv")

df.head()

name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  ...        D2       PPE  status
0  phon_R01_S01_1      119.992       157.302  ...  2.301442  0.284654       1
1  phon_R01_S01_2      122.400       148.650  ...  2.486855  0.368674       1
2  phon_R01_S01_3      116.682       131.111  ...  2.342259  0.332634       1
3  phon_R01_S01_4      116.676       137.871  ...  2.405554  0.368975       1
4  phon_R01_S01_5      116.014       141.781  ...  2.332180  0.410335       1

[5 rows x 24 columns]

### **x represents Features and y represents Label / Target**

In [52]:
x = df.iloc[ : , 1: -1].values

y = df.iloc[ : , -1].values

In [53]:
print(x)

print(y)

[[1.199920e+02 1.573020e+02 7.499700e+01 ... 2.664820e-01 2.301442e+00
  2.846540e-01]
 [1.224000e+02 1.486500e+02 1.138190e+02 ... 3.355900e-01 2.486855e+00
  3.686740e-01]
 [1.166820e+02 1.311110e+02 1.115550e+02 ... 3.111730e-01 2.342259e+00
  3.326340e-01]
 ...
 [1.746880e+02 2.400050e+02 7.428700e+01 ... 1.584530e-01 2.679772e+00
  1.317280e-01]
 [1.987640e+02 3.969610e+02 7.490400e+01 ... 2.074540e-01 2.138608e+00
  1.233060e-01]
 [2.142890e+02 2.602770e+02 7.797300e+01 ... 1.906670e-01 2.555477e+00
  1.485690e-01]]
[1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 1
 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 0 0 0 0 0 0 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 0 0
 0 0 0 0 0 0 0 0 0 0]


### **to identify the number of instances suffering and not suffering from Parkinson's disease as per the given data**

In [54]:
print(y[y==1].shape[0] , y[y==0].shape[0])

147 48


### **MinMaxScalar : to normalise ie. to bring the values within a particular range**

In [55]:
from sklearn.preprocessing import MinMaxScaler
scaling = MinMaxScaler((-1,1))
x = scaling.fit_transform(x)

In [56]:
print(x)

[[-0.63138346 -0.77481654 -0.89037042 ...  0.17153026 -0.21867743
  -0.0053808 ]
 [-0.6033463  -0.81013911 -0.4433544  ...  0.48267409 -0.05370956
   0.34265204]
 [-0.66992292 -0.88174367 -0.46942324 ...  0.37274182 -0.18236124
   0.19336492]
 ...
 [ 0.00546073 -0.43717403 -0.89854572 ... -0.31484696  0.11793486
  -0.63884033]
 [ 0.28578581  0.20361309 -0.89144127 ... -0.09423055 -0.36355605
  -0.67372646]
 [ 0.46654868 -0.35441175 -0.85610326 ... -0.16981039  0.00734563
  -0.5690805 ]]


### **train_test_split : to decide the data to be considered for training and testing**

In [57]:
from sklearn.model_selection import train_test_split

x_train , x_test , y_train, y_test = train_test_split(x,y,test_size=0.2,random_state=7)

### **Model Training / Fitting : to train the model using XGBoost classification algorithm**

In [58]:
from xgboost import XGBClassifier
model = XGBClassifier()

model.fit(x_train,y_train)

XGBClassifier()

### **Model Testing : prediction of Parkinson's disease**

In [59]:
y_pred = model.predict(x_test)

### **Comparison between y_test and y_pred :** 
### **to compare between the actual truth / diagnosis and the prediction of the Parkinson's disease**

In [60]:
comparison = pd.DataFrame({"True Report" : y_test , "Predicted Report" : y_pred})

In [61]:
comparison.head()

True Report  Predicted Report
0            0                 1
1            1                 1
2            1                 1
3            1                 1
4            1                 1

### **To verify the accuracy of the model by the following metrics:**

### 1.   Confusion Matrix
### 2.   Accuracy Score
### 3.   Recall Score
### 4.   Precision Score
### 5.   F1 Score
### 6.   AUC ROC Score

In [62]:
from sklearn import metrics

# 1.   Confusion Matrix
print(metrics.confusion_matrix(y_test,y_pred))

# 2.   Accuracy Score
print(metrics.accuracy_score(y_test,y_pred) * 100)

# 3.   Recall Score
print(metrics.recall_score(y_test,y_pred))

# 4.   Precision Score
print(metrics.precision_score(y_test,y_pred)*100)

# 5.   F1 Score
print(metrics.f1_score(y_test,y_pred)*100)

# 6.   AUC ROC Score
print(metrics.roc_auc_score(y_test,y_pred)*100)

[[ 5  2]
 [ 0 32]]
94.87179487179486
1.0
94.11764705882352
96.96969696969697
85.71428571428572


## Hence, the model has an accuracy score of 94.87 %